pkg/cloud/azure/actuators/machine/reconciler.go

```diff
@@ -470,9 +470,17 @@ func (r *Reconciler) createVirtualMachine(ctx context.Context, nicName string) e
 
 	vmInterface, err := r.virtualMachinesSvc.Get(ctx, vmSpec)
 	if err != nil && vmInterface == nil {
-		vmZone, zoneErr := r.getVirtualMachineZone(ctx)
-		if zoneErr != nil {
-			return errors.Wrap(zoneErr, "failed to get availability zone")
+		var vmZone string
+		var zoneErr error
+
+		vmZone = r.scope.MachineConfig.AZ
+
+		if r.scope.MachineConfig.AZ == "" {
+			klog.Info("No availability zone set, seletecting random availability zone")
+			vmZone, zoneErr = r.getVirtualMachineZone(ctx)
```

Comment:
zoneErr could be declared above here, reducing its scope

Reply:
I'm not sure what you mean here. zoneErr is declared just above this (line 495 after rebase)

Reply:
I was suggesting to add a line before here like `var err error` to reduce the scope of the error :). Not really necessary, just a little nit because the error isn't actually used anywhere else within the function